python/cudf/cudf/core/mixins/mixin_factory.py

```diff
@@ -93,49 +104,46 @@ def _create_delegating_mixin(
     >>> print(bar.foo1.__doc__)
     Perform the operation foo1, which returns 42.
     """
-
     docstring_attr = f"{category_name}_DOCSTRINGS"
     validity_attr = f"_VALID_{category_name}S"
 
-    class OperationMixin:
-        @classmethod
-        def _add_operation(cls, operation):
-            # This function creates operations on-the-fly.
-
-            # Generate a signature without the `op` parameter.
-            base_operation = getattr(cls, operation_name)
-            base_params = inspect.signature(base_operation).parameters.values()
-            params = [p for p in base_params if p.name != "op"]
-            signature = inspect.Signature(parameters=params)
-
-            # Generate the list of arguments forwarded to the base operation.
-            arglist = ", ".join(
-                [
-                    f"{key}={key}"
-                    for key in signature.parameters
-                    if key not in ("self", "args", "kwargs")
-                ]
-            )
-            arglist = arglist + (", " if arglist else "") + "*args, **kwargs"
-
-            # Apply the formatted docstring of the base operation to the
-            # operation being created here.
-            docstring = base_operation.__doc__.format(
-                cls=cls.__name__,
-                op=operation,
-                **getattr(cls, docstring_attr, {}).get(operation, {}),
-            )
+    class Operation:
+        """
+        """
+
+        def __init__(self):
+            self._operation_name = operation_name
```

Comment:
This is the name of the "base" operation, i.e., `_reduce`.

Reply:
Do we even need an `__init__`? We're binding the variable from the closure anyway, so shouldn't it work just fine to just use `operation_name` inside `__get__`? I don't think there are any late binding issues here.

Reply:
You're right -- we can remove it.